packages/bug-detectors/command-injection.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2023 Code Intelligence GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+	BugDetectorError,
+	hookBuiltInFunction,
+	saveFirstBugDetectorError,
+} from "./index";
+import { nextFakePC } from "@jazzer.js/instrumentor/dist/plugins/helpers";
+import { guideTowardsEquality } from "@jazzer.js/fuzzer";
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export async function registerCommandInjectionBugDetectors<F extends Function>(
+	callOriginalFn: boolean
+): Promise<void> {
+	async function registerBugDetector(
+		moduleName: string,
+		targetFunctionName: string,
+		evilCommand: string,
+		baseErrorString: string,
+		callOriginalFn: boolean
+	): Promise<F> {
+		const id = nextFakePC();
+		return await hookBuiltInFunction(
+			moduleName,
+			targetFunctionName,
+			(
+				originalFn: F,
+				cmdOrFileOrPath: string,
+				...args: unknown[]
+			): F | void => {
+				guideTowardsEquality(cmdOrFileOrPath, evilCommand, id);
+				if (cmdOrFileOrPath.includes(evilCommand)) {
+					const err = new BugDetectorError(
+						baseErrorString +
+							targetFunctionName +
+							"() called with command: '" +
+							cmdOrFileOrPath +
+							"'"
+					);
+					// Remove the first 3 lines after the message from the stack trace.
+					// The first lines are internal Jazzer function calls.
+					saveFirstBugDetectorError(err, 3);
+					throw err;
+				}
+				if (callOriginalFn) {
+					return originalFn(cmdOrFileOrPath, ...args);
+				}
+			}
+		);
+	}
+	const evilCommand =
```

Comment:
We could just check if an expected string, something small and probably unused like `jazze` or `jazCom`, reaches one of the listed functions. If that's the case fuzzer input is used to execute commands. Shouldn't that be sufficient for this bug detector?  This would also remove the need to distinguish the OS.

Reply:
Changed the special command to 'jaz_zer'.

All other commands should definitely run, so I use "touch FRIENDLY" / "copy NUL FRIENDLY" and check that a file is created. If you know of an OS-agnostic command I could use instead, let me know!